### PR TITLE
Fixed the RallyPoint trait for Tiberian Sun polish

### PR DESCRIPTION
**Pull request by Mailaender**
The sprites are not hard-coded, but configurable, self-documented and partly optional. Makes it lint testable regarding sprites and also future palette references https://github.com/OpenRA/OpenRA/pull/8749 as well.

### CODE REVIEW

**Comment by Phrohdoh** [comment]
The point in our `Desc` attribute is to provide descriptions even in the case of self-documentation.

File: OpenRA.Mods.Common/Effects/RallyPointIndicator.cs [hunk 1]
```diff
@@ -20,33 +20,44 @@ class RallyPointIndicator : IEffect
...
+		readonly string paletteName;
 		readonly Animation flag;
 		readonly Animation circles;
 
-		public RallyPointIndicator(Actor building, string palettePrefix)
+		public RallyPointIndicator(Actor building, string palette)
 		{
 			this.building = building;
-			this.palettePrefix = palettePrefix;
+			this.paletteName = palette;
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Effects/RallyPointIndicator.cs]: I don't think you need the `this.` here anymore.

**Comment by abcdefg30** [comment]
TS looks good. But in RA are the circles gone:
rally-nocircle

**Comment by Mailaender** [comment]
Woops, reverted some parts to fix this.

**Comment by abcdefg30** [comment]
Looks good to me now. :+1: / :white_check_mark:

**Comment by RoosterDragon** [comment]
:+1: Needs a rebase.

**Comment by Mailaender** [comment]
Rebased.

**Pull request merged**